packages/worker/src/worker.ts

```diff
@@ -1060,7 +1081,12 @@ export class Worker {
             return { taskToken, result };
           }),
           filter(<T>(result: T): result is Exclude<T, undefined> => result !== undefined),
-          map((rest) => coresdk.ActivityTaskCompletion.encodeDelimited(rest).finish())
+          map((rest) => coresdk.ActivityTaskCompletion.encodeDelimited(rest).finish()),
+          tap({
+            next: () => {
+              group$.close();
+            },
```

Comment:
A comment about why it is not enough to close() in the finally{} would be useful here...

Reply:
Simply because we may never go through that try-finally block... 😉

That might not be obvious by looking at the diff just now, but there are a few code paths above that throws or returns.